packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -1072,7 +1072,9 @@ function collectParams(
       if (value.type.kind === "Model") {
         collectParams(value.type.properties, params);
       } else {
-        params.push(value);
+        if (!isApiVersion(value)) {
```

Comment:
ideally, I want to check if this is from a core parameter. This is hard, I tried doing `isAzureCoreModel(value.sourceModel.model)`, but the issue is it seems to be catching even `foo` to be of an `Azure.Core` model. @tadelesh do you have any recs?

Reply:
`isAzureCoreModel` function miss one logic: it should filter all templated instances model which belong to the service but with core namespace.

Reply:
ok let me try it this way today